data_steward/tools/create_tier.py

```diff
@@ -81,6 +83,112 @@ def get_dataset_name(tier, release_tag, deid_stage):
     return dataset_name
 
 
+def create_datasets(client, name, input_dataset, tier, release_tag):
+    """
+    Creates backup, staging, sandbox, and final datasets with the proper descriptions
+    and tag/labels applied
+
+    :param client: an instantiated bigquery client object
+    :param name: the base name of the datasets to be created
+    :param input_dataset: name of the input dataset
+    :param tier: tier parameter passed through from either a list or command line argument
+    :param release_tag: release tag parameter passed through either the command line arguments
+    :return: tuple of created dataset names
+    """
+
+    if not client:
+        raise RuntimeError("Please specify BigQuery client object")
+    if not name:
+        raise RuntimeError(
+            "Please specify the base name of the datasets to be created")
+    if not input_dataset:
+        raise RuntimeError("Please specify the name of the input dataset")
+    if not tier:
+        raise RuntimeError(
+            "Please specify the tier intended for the output datasets")
+    if not release_tag:
+        raise RuntimeError(
+            "Please specify the release tag for the dataset in the format of YYYY#q#r"
+        )
+
+    # Construct names of datasets need as part of the deid process
+    final_dataset_id = name
+    backup_dataset_id = f'{name}_{consts.BACKUP}'
+    staging_dataset_id = f'{name}_{consts.STAGING}'
+    sandbox_dataset_id = f'{name}_{consts.SANDBOX}'
+
+    dataset_ids = [
+        final_dataset_id, backup_dataset_id, staging_dataset_id,
+        sandbox_dataset_id
+    ]
+
+    # base labels and tags for the datasets
+    base_labels_and_tags = {'release_tag': release_tag, 'data_tier': tier}
+
+    description = f'dataset created from {input_dataset} for {tier}{release_tag} CDR run'
+
+    # Creation of dataset objects and dataset label and description updates
+    for dataset in dataset_ids:
+        dataset_object = bq.define_dataset(client.project, dataset, description,
+                                           base_labels_and_tags)
+        client.create_dataset(dataset_object, exists_ok=True)
+
+        if dataset is sandbox_dataset_id:
+            new_labels = bq.update_labels_and_tags(sandbox_dataset_id,
+                                                   base_labels_and_tags,
+                                                   {'phase': consts.SANDBOX})
+            dataset = bq.get_dataset(client.project, sandbox_dataset_id)
+            dataset.labels = new_labels
+            dataset.description = f'{consts.SANDBOX} {description}'
+            client.update_dataset(dataset, ["labels", "description"])
```

Comment:
It looks like the loop variables can be simplified into one block without the if, using `datasets` defined below:
```suggestion
    for phase, dataset_id in datasets.items():
        dataset_object = bq.define_dataset(client.project, dataset_id, description,
                                           base_labels_and_tags)
        client.create_dataset(dataset_object, exists_ok=True)

        new_labels = bq.update_labels_and_tags(dataset_id, base_labels_and_tags,
                                               {'phase': phase})
        dataset = bq.get_dataset(client.project, dataset_id)
        dataset.labels = new_labels
        dataset.description = f'{phase} {description}'
        client.update_dataset(dataset, ["labels", "description"])
```

Reply:
> Added one small suggestion. Is the failing unit test related to the PR changes? I'm not sure why it is failing.

No, it's not related to the PR, I didn't change anything in `data_steward.utils.bq_test.BqTest` where it is stemming from.

Reply:
Lets have a convo tomorrow about your suggestions, I am confused at where the `datasets` list is coming from. I only have the list of `dataset_ids`

Reply:
`datasets` is the output dictionary the function returns. It is currently defined below, but it would have to be moved above this loop.

Reply:
Ah, gotcha!

Reply:
I may still need an if statement to add de-identified tag to the staging and final datasets

Reply:
You're right, a variable like deidentified = 'true' if phase in [consts.STAGING, consts.CLEAN] else 'false' could help

Reply:
I ended up making a list of `deid_datasets` that contains `final_dataset_id` and `staging_dataset_id` and if the dataset_id is in that list then it will add the `'de-identified': 'true'` else `'de-identified': 'false'`. It looks much cleaner than what I had before.